x-pack/plugins/alerting/server/types.ts

```diff
@@ -207,6 +199,10 @@ interface ComponentTemplateSpec {
   fieldMap: FieldMap;
 }
 
+export type FormatAlert = <AlertData extends RuleAlertData>(
+  alert: Alert & AlertData
+) => DeepPartial<Alert & AlertData>;
+
```

Comment:
`alert: Partial<AlertData>` and  `Partial<AlertData>;`  as a return type could be enough...